star-dash/star-dash/Rendering/MTKRenderer/JoystickView.swift

```diff
@@ -0,0 +1,74 @@
+import UIKit
+
+/**
+ `JoystickView` is represents the joystick which the user drags with
+ to move the character.
+ */
+class JoystickView: UIView {
```

Comment:
Not sure if this https://developer.apple.com/documentation/gamecontroller/gcvirtualcontroller is helpful, could be a better way to do the controller with the additional support for external game controller